### ISSUE TITLE
Project's files list with branchId filter causes an empty result

### ISSUE THREAD

**Issue by erickskrauch** [crowdin/crowdin-api-client-js#63]
Hello. Calling `crowdin.sourceFilesApi.listProjectFiles()` function with the second argument (`branchId`) causes an empty result (branch exists). Am I doing something wrong or it's expected behavior?

Curl request:
```sh
curl -X GET 'https://api.crowdin.com/api/v2/projects/350687/files?branchId=26' \
    -H 'Authorization: Bearer *masked*'
```

Response:
```json
{
    "data": [],
    "pagination": {
        "offset": 0,
        "limit": 25
    }
}
```

Without the `branchId` param, the response contains the necessary file and you can see, that `branchId` is correct.

```json
{
    "data": [
        {
            "data": {
                "revisionId": 5,
                "status": "active",
                "priority": "high",
                "importOptions": null,
                "exportOptions": null,
                "createdAt": "2019-03-04T12:26:21+00:00",
                "updatedAt": "2020-05-23T23:20:48+00:00",
                "revision": 5,
                "id": 6,
                "projectId": 350687,
                "branchId": null,
                "directoryId": 4,
                "name": "site.json",
                "title": null,
                "type": "json"
            }
        },
        {
            "data": {
                "revisionId": 1,
                "status": "active",
                "priority": "normal",
                "importOptions": null,
                "exportOptions": null,
                "createdAt": "2020-06-08T23:25:30+00:00",
                "updatedAt": "2020-06-08T23:25:31+00:00",
                "revision": 1,
                "id": 32,
                "projectId": 350687,
                "branchId": 26,
                "directoryId": 30,
                "name": "site.json",
                "title": null,
                "type": "json"
            }
        }
    ],
    "pagination": {
        "offset": 0,
        "limit": 25
    }
}
```

**Comment by Olena1234** [comment]
Hi @erickskrauch ! We will check this behavior with the developers and come back to you with an answer as soon as they are in the office! Thank you

**Comment by Olena1234** [comment]
Hi @erickskrauch ! Just following up and wanted to let you know that task on the matter is already created and developers are looking into it. We will let you know as soon as it's done!

**Comment by eskimos2008** [comment]
Hi @erickskrauch ! It is expected behaviour. it happens because your file is located in subdirectory. if you want to see all child files of the branch, use 'recursion' query parameter in request.

**Comment by erickskrauch** [comment]
@eskimos2008, thanks for the reply. The problem is that the `sourceFilesApi.listProjectFiles` has no ability to call the list recursively:
https://github.com/crowdin/crowdin-api-client-js/blob/6ac19486ea8df2b4ac28260f515961608cdd9579/src/sourceFiles/index.ts#L133-L154

I suggest replacing the signature to `projectId: number, request?: SourceFilesModel.ListProjectFilesRequest`. I can send a PR if necessary.

**Comment by yevheniyJ** [comment]
@erickskrauch hi there! No need, thanks) I will take care of it. And just for backward compatibility I will add one more argument to method.

**Comment by erickskrauch** [comment]
I suggest introducing overloading rather than adding one another method: if the second argument is an object, than take options from it.

**Comment by yevheniyJ** [comment]
Yep, overloading make sense here.

**Comment by yevheniyJ** [comment]
PR #66

**Comment by yevheniyJ** [comment]
Now available in 1.8.6
